src/sentry/taskworker/taskworker.py

```diff
@@ -0,0 +1,190 @@
+from __future__ import annotations
+
+import logging
+import time
+from datetime import datetime
+from multiprocessing.context import TimeoutError
+from multiprocessing.pool import Pool
+from uuid import uuid4
+
+import grpc
+import orjson
+from django.conf import settings
+from sentry_protos.sentry.v1.taskworker_pb2 import (
+    TASK_ACTIVATION_STATUS_COMPLETE,
+    TASK_ACTIVATION_STATUS_FAILURE,
+    TASK_ACTIVATION_STATUS_RETRY,
+    TaskActivation,
+)
+
+from sentry.taskworker import taskworker_process
+from sentry.taskworker.registry import TaskNamespace, taskregistry
+
+logger = logging.getLogger("sentry.taskworker")
+result_logger = logging.getLogger("taskworker.results")
+
+
+class WorkerCompleteException(Exception):
+    pass
+
+
+class TaskWorker:
+    """
+    A TaskWorker fetches tasks from a task queue (inflight actiivation store) and handles executing them.
+    As a tasks are executed, the worker will update the task status in the task queue.
+    """
+
+    def __init__(self, **options):
+        self.options = options
+        self.exitcode = None
+        self._execution_count = 0
+        self._worker_id = uuid4().hex
+        self._namespace: TaskNamespace | None = None
+        self._build_pool()
+
+    def _build_pool(self) -> None:
+        self._pool = Pool(processes=2)
+
+    @property
+    def namespace(self) -> TaskNamespace:
+        if self._namespace:
+            return self._namespace
+
+        name = self.options["namespace"]
+        self._namespace = taskregistry.get(name)
+        return self._namespace
+
+    def do_imports(self) -> None:
+        for module in settings.TASKWORKER_IMPORTS:
+            __import__(module)
+
+    def start(self) -> None:
+        self.do_imports()
+        max_task_count = self.options.get("max_task_count", None)
+        next_task = None
+        processing_deadline = None
+        try:
+            while True:
+                if next_task:
+                    task = next_task
+                    next_task = None
+                else:
+                    (task, processing_deadline) = self.fetch_task()
+
+                if not task:
+                    time.sleep(1)
+                    continue
+
+                next_task, processing_deadline = self.process_task(task, processing_deadline)
+
+                if max_task_count is not None and max_task_count <= self._execution_count:
+                    self.exitcode = 1
+                    raise WorkerCompleteException("Max task exeuction count reached")
+
+        except KeyboardInterrupt:
+            self.exitcode = 1
+        except WorkerCompleteException as err:
+            self.exitcode = 0
+            logger.warning(str(err))
```

Comment:
I feel like it might be easier to maintain if we just return the exit code here, instead of creating a custom exception and catching it
```python
if max_task_count is not None and max_task_count <= self._execution_count:
    logger.warning("Max task execution count reached")
    return 1
```